packages/config/src/config.ts

```diff
@@ -92,14 +92,29 @@ export const addressOf = (salt: WalletConfig | string, context: WalletContext, i
   return addressOf(imageHash(salt), context)
 }
 
+// cached wallet configs indexed by image hash
+// wallet configs must be sorted using sortConfig
+const cachedConfigs: Map<string, WalletConfig> = new Map()
+
```

Comment:
maybe we should add another method called, `setCachedConfig(xxx)` so that we could get/set this from the wallet-webapp from local storage, so it would survive reloads?

in other parts in findConfig, if we load something from cache, we should check still the on-chain "imageHash" compared to our latest cached value, this way we'll know if our cache is expired..

what do you think?

Reply:
It's possible to move this cache from memory to local storage, but in that case, we will have to consider adding an eviction strategy so that the cache isn't unbounded.

> in other parts in findConfig, if we load something from cache, we should check still the on-chain "imageHash" compared to our latest cached value, this way we'll know if our cache is expired..

I think I might be misunderstanding this remark. `findConfigForImageHash` has a specific purpose, to find the config for a specific image hash. If you're referring to `findCurrentConfig`, then it already makes a request for the on-chain image hash.